src/Microsoft.TemplateEngine.Edge/BuiltInManagedProvider/GlobalSettings.cs

```diff
@@ -157,6 +163,75 @@ public async Task SetInstalledTemplatePackagesAsync(IReadOnlyList<TemplatePackag
             throw new InvalidOperationException();
         }
 
+        public async Task MigrateCacheMetadataAsync(CancellationToken cancellationToken)
+        {
+            if (!CanApplyFirstRunDataMigration())
+            {
+                return;
+            }
+
+            _logger.LogWarning(LocalizableStrings.GlobalSettingsTemplatePackageProvider_DataMigrationWarning);
+
+            string sourceFeedKey = "NuGetSource";
+            string packageIdKey = "PackageId";
+            string versionKey = "Version";
+            string ownersKey = "Owners";
+            string trustedKey = "Trusted";
```

Comment:
It's better to use those https://github.com/dotnet/templating/blob/6536e8d9b4ae1e70047cf4c254c352db75b99903/src/Microsoft.TemplateEngine.Edge/Installers/NuGet/NuGetManagedTemplatePackage.cs#L16-L22, just make them internal.